ui/src/app/views/account-creation/component.ts

```diff
@@ -145,4 +167,83 @@ export class AccountCreationComponent {
     return (/[A-Z]/.test(str));
   }
 
+  leaveFocusUsername(): void {
+    this.usernameOffFocus = true;
+  }
+
+  enterFocusUsername(): void {
+    this.usernameOffFocus = false;
+  }
+
+  get isUsernameValidationError(): boolean {
+    return this.usernameConflictError || this.usernameInvalidError;
+  }
+
+  get showUsernameValidationError(): boolean {
+    if (isBlank(this.profile.username) || !this.usernameOffFocus) {
+      return false;
+    }
+    return this.isUsernameValidationError;
+  }
+
+  get showUsernameValidationSuccess(): boolean {
+    if (isBlank(this.profile.username) || !this.usernameOffFocus) {
+      return false;
+    }
+    return !this.isUsernameValidationError;
+  }
+
+  leaveFocusPassword(): void {
+    this.passwordOffFocus = true;
+  }
+
+  enterFocusPassword(): void {
+    this.passwordOffFocus = false;
+  }
+
+  get isPasswordValidationError(): boolean {
```

Comment:
nit: rereading this, I might invert and rename a bit, maybe `passwordIsValid`? (here and for passwordAgain)

Reply:
I like having it be true when its invalid here, but I can change to `passwordIsInvalid`?